shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -369,9 +375,16 @@ bool FlutterWindowsEngine::Run(std::string_view entrypoint) {
     args.aot_data = aot_data_.get();
   }
 
-  FlutterRendererConfig renderer_config = surface_manager_
-                                              ? GetOpenGLRendererConfig()
-                                              : GetSoftwareRendererConfig();
+  FlutterRendererConfig renderer_config;
+
+  if (enable_impeller_) {
+    // Impeller does not support a Software backend. Avoid failling back and
+    // confusing the engine on which renderer is selected.
+    renderer_config = GetOpenGLRendererConfig();
```

Comment:
Should we `return false` here if surface creation failed and the `surface_manager_` is null?

Reply:
If yes, I'd also test this behavior here using something similar to this: https://github.com/flutter/engine/blob/1de141c6a6e744b30a1de179370aa0805e4e7b94/shell/platform/windows/flutter_windows_engine_unittests.cc#L173

Reply:
Done